subsys/net/ip/net_context.c

```diff
@@ -368,23 +368,19 @@ static int check_used_port(enum net_ip_protocol ip_proto,
 static u16_t find_available_port(struct net_context *context,
 				    const struct sockaddr *addr)
 {
-	if (!net_sin(addr)->sin_port) {
-		u16_t local_port;
```

Comment:
Wasn't the point of this so the user can actually select the port he wants? I remember @rveerama1  working on this.

If it happens to be uninitialized, then the bug is elsewhere: it should never be uninitialized, default should always be 0 then. no?

Reply:
@tbursztyka : As I mentioned, that was my concern. User can "select port he wants" using a call to bind(), and bind_default() checks for "already bound", albeit in a particular way, we really need to add net_context_is_bound() call. So, everything seems to work ok. (And call of find_available_port() in net_context_get() is superfluous, yes.)

Reply:
The find_available_port() is called from net_context_get(), and there the port value is always initialized to 0. The other call is from bind_default() where the port is left uninitialized even thou it should be initialized. But this uninitialization does not really matter here as the return value of find_available_port() is used as a new port number anyway.
To my eyes it makes sense to fix find_available_port() to return a new random port number which is then always used.

Reply:
Indeed, double checking this: net_context_bind() use check_used_port()  in net_context.c:806

Ok then.